openshift-hack/e2e/annotate/annotate.go

```diff
@@ -110,25 +123,34 @@ func init() {
 	}
 }
 
+type matchable struct {
+	pattern string
+	literal string
+	re      *regexp.Regexp
+	matched bool
+}
+
 func newGenerator(testMaps map[string][]string) *ginkgoTestRenamer {
 	var allLabels []string
-	matches := make(map[string]*regexp.Regexp)
-	stringMatches := make(map[string][]string)
+	matches := make(map[string][]*matchable)
 
 	for label, items := range testMaps {
 		sort.Strings(items)
 		allLabels = append(allLabels, label)
-		var remain []string
 		for _, item := range items {
+			if _, exists := matches[item]; exists {
```

Comment:
Don't we index matches by the label and not the item/pattern?

Can you please introduce simple tests for the new logic?

Reply:
@danwinship can you please address this point?

Reply:
yes, that check was incorrect, and I don't remember why I'd originally added it so I just removed it.